client/code-mirror.js

```diff
@@ -87,9 +91,12 @@ var CodeMirror = React.createClass({
         }
       }
       console.log(filename)
-      api.uploadImage(event.target.result, filename).then((res) =>
-        this.cm.replaceSelection(`\n![${res.msg}](${res.src})`)
-      );
+      api.uploadImage(event.target.result, filename).then((res) => {
+        setTimeout(() => {
+          // waiting for write to disk.
+          this.cm.replaceSelection(`\n![${res.msg}](${res.src})`)
+        }, 1000);
```

Comment:
I'd rather have the `uploadImage` request just wait until the image is written to disk instead of having an arbitrary timeout